### PR TITLE
Add typing stubs to pre-commit mypy run

### PR DESCRIPTION
**Pull request by akkomar**
This is a follow-up to https://mozilla.slack.com/archives/C01E8GDG80N/p1665434432654689

We've seen mypy pre-commit hook failing and complaining about missing library stubs for some of the dependencies. Per [pre-commit mirror docs](https://github.com/pre-commit/mirrors-mypy#using-mypy-with-pre-commit) it runs in an isolated environment so it seems to be making sense.

The issue might be related to https://github.com/mozilla/bigquery-etl/pull/3248.

Changes proposed here do fix the mypy hook run for me on Python 3.10.7. However since I don't understand why we're seeing these errors I am not entirely sure if the local environment (mine and @kik-kik ) isn't at fault here.

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

**Review by akkomar — COMMENTED** [review]
Lint

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -32,3 +32,9 @@ repos:
   rev: v0.982
   hooks:
   - id: mypy
+    additional_dependencies: [types-pytz==2022.4.0.0,
+                              types-ujson==5.5.0,
+                              types-python-dateutil==2.8.19,
+                              types-requests==2.28.11.2,
+                              types-attrs==19.1.0,
+                              types-PyYAML==6.0.12]
```
Comment (akkomar) [inline .pre-commit-config.yaml:40]: These are actually duplicates of values from requirements.in which is not great.

**Comment by kik-kik** [comment]
I can confirm using this config solves the errors on my machine

File: .pre-commit-config.yaml (hunk 1 above)
Comment (kik-kik) [reply .pre-commit-config.yaml:40]: looks like we need to specify the additional_dependencies...

**Review by relud — APPROVED** [review]

**Pull request merged**